### ISSUE TITLE
undefined method `encoding' for 0:Fixnum

### ISSUE THREAD

**Issue by henriquemenezes** [timberio/timber-ruby#91]
The timber gem throws this exception in Rails application.

MESSAGE

`NoMethodError: undefined method `encoding' for 0:Fixnum`

CODE

`timber/util/string.rb in normalize_to_utf8 at line 9`

```ruby
      # @private
      def self.normalize_to_utf8(string)
        if string.encoding.to_s == UTF8
          string
        else
          string.encode('UTF-8', {
```

EXCEPTION

```
NoMethodError: undefined method `encoding' for 0:Fixnum
  from timber/util/string.rb:9:in `normalize_to_utf8'
  from timber/util/http_event.rb:31:in `block in normalize_headers'
  from timber/util/http_event.rb:27:in `each'
  from timber/util/http_event.rb:27:in `each_with_object'
  from timber/util/http_event.rb:27:in `normalize_headers'
  from timber/events/http_server_response.rb:12:in `initialize'
  from timber/integrations/rack/http_events.rb:31:in `new'
  from timber/integrations/rack/http_events.rb:31:in `block in call'
  from logger.rb:427:in `add'
  from timber/logger.rb:233:in `info'
  from timber/integrations/rack/http_events.rb:29:in `call'
  from timber/integrations/rack/user_context.rb:21:in `call'
  from timber/integrations/rack/session_context.rb:18:in `call'
  from timber/integrations/rack/http_context.rb:19:in `block in call'
  from timber/current_context.rb:65:in `with'
  from timber/current_context.rb:18:in `with'
  from timber/integrations/rack/http_context.rb:18:in `call'
  from rack/etag.rb:25:in `call'
  from rack/conditional_get.rb:25:in `call'
  from rack/head.rb:12:in `call'
  from action_dispatch/middleware/callbacks.rb:38:in `block in call'
  from active_support/callbacks.rb:97:in `__run_callbacks__'
  from active_support/callbacks.rb:750:in `_run_call_callbacks'
  from active_support/callbacks.rb:90:in `run_callbacks'
  from action_dispatch/middleware/callbacks.rb:36:in `call'
  from action_dispatch/middleware/remote_ip.rb:79:in `call'
  from action_dispatch/middleware/debug_exceptions.rb:49:in `call'
  from action_dispatch/middleware/show_exceptions.rb:31:in `call'
  from rails/rack/logger.rb:36:in `call_app'
  from rails/rack/logger.rb:26:in `call'
  from action_dispatch/middleware/request_id.rb:24:in `call'
  from rack/runtime.rb:22:in `call'
  from active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
  from action_dispatch/middleware/executor.rb:12:in `call'
  from rack/sendfile.rb:111:in `call'
  from raven/integrations/rack.rb:50:in `call'
  from rack/cors.rb:80:in `call'
  from rails/engine.rb:522:in `call'
  from puma/configuration.rb:225:in `call'
  from puma/server.rb:578:in `handle_request'
  from puma/server.rb:415:in `process_client'
  from puma/server.rb:275:in `block in run'
  from puma/thread_pool.rb:116:in `block in spawn_thread'
```

**Comment by binarylogic** [comment]
Ah crap. I'll get a test in place and release a patch update today. Thanks for reporting.

**Comment by binarylogic** [comment]
Ok, `2.0.24` was released and resolves this. Also, we'll be released `2.1.0` sometime next week that introduces a number of great features and updates.
